### ISSUE TITLE
BUG: v0.1.3 parseMultipartBody fails 

### ISSUE THREAD

**Issue by kappelmann** [silkapp/multipart#4]
GIven the following program
```haskell
import Network.Multipart
import Data.ByteString.Lazy.UTF8 as BLU

main = let boundary = "---------------------------9051914041544843365972754266" in
    putStr $ show $ parseMultipartBody boundary $ BLU.fromString "-----------------------------9051914041544843365972754266\n\
\Content-Disposition: form-data; name=\"text\"\n\
\\n\
\text default\n\
\-----------------------------9051914041544843365972754266\n\
\Content-Disposition: form-data; name=\"file1\"; filename=\"a.txt\"\n\
\Content-Type: text/plain\n\
\\n\
\Content of a.txt.\n\
\-----------------------------9051914041544843365972754266--"
```
the output with v0.1.3 is
```haskell
MultiPart []
```
Changing multipart to v0.1.2 changes the output to
```haskell
MultiPart [BodyPart [(HeaderName "Content-Disposition","form-data; name=\"text\"")] "text default",BodyPart [(HeaderName "Content-Disposition","form-data; name=\"file1\"; filename=\"a.txt\""),(HeaderName "Content-Type","text/plain")] "Content of a.txt."]
```

**Comment by hesselink** [comment]
@ali-abrar Any idea what's happening here? The only change between those versions was your performance improvement.

**Comment by cheecheeo** [comment]
as far as I can tell this bug is also affecting version 0.2.0.

https://travis-ci.org/cheecheeo/haskell-cgi/jobs/655134595#L557
https://travis-ci.org/cheecheeo/haskell-cgi/jobs/655134595#L591

**Comment by cescobaz** [comment]
I have the same issue in version 0.1.3 and 0.2.0.
0.1.2 works fine.
I noticed that it always skips the first part of multiparts then keeps the others.

**Comment by matthewbauer** [comment]
This should be addressed in https://github.com/silkapp/multipart/pull/8.

**Comment by snakamura** [comment]
@hesselink Are there any chances #8 gets merged?

**Comment by gzh** [comment]
I can confirm that this PR successfully fixes parsing of multipart response for data like [multipart-examples.zip](https://github.com/silkapp/multipart/files/5391399/multipart-examples.zip).
The files in that archive are produced by Pelco VideoXpert VMS and are supposed to contain JPEG images (1 in one file, and 8 in another). The multipart-0.1.3 fails to parse the file with 1 image, and extracts 7 images from the file there are supposed to be 8. The revision 7e0575489f8ab8621b01079fb513fa1d365eef11 works perfectly for this examples of multipart data.

**Comment by hesselink** [comment]
The linked PR has been merged and released as 0.2.1. Thanks everyone for contributing and testing, and sorry this took a while to get merged.
